### PR TITLE
Fix for Specify IFormatProvider errors

### PR DESCRIPTION
**Pull request by ngastelum-ms**
[CA1305: Specify IFormatProvider](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1305)

### CODE REVIEW

File: iothub/device/src/Transport/Mqtt/MqttLogger.cs [hunk 1]
```diff
@@ -21,7 +22,7 @@ public void Publish(MqttNetLogLevel logLevel, string source, string message, obj
             string formattedMessage = message;
             if (parameters != null && parameters.Length > 0)
             {
-                formattedMessage = string.Format(message, parameters);
+                formattedMessage = string.Format(CultureInfo.CurrentCulture, message, parameters);
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/Mqtt/MqttLogger.cs:25]: I can see the point that a log statement should be formatted based on current culture; however, I wanted to get this confirmed from what the track 2 SDKs do.
Turns out, they format based on InvariantCulture, possibly to preserve the logging behavior across regions.

I'd recommend that we do the same.

https://github.com/Azure/azure-sdk-for-net/blob/e59553b4fb61b37c9c13681b4b117cc58de17e6d/sdk/core/Azure.Core/src/Diagnostics/AzureEventSourceListener.cs#L96

File: iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs [hunk 2]
```diff
@@ -1239,7 +1239,7 @@ private bool ParseResponseTopic(string topicName, out string rid, out int status
                 {
                     // This query string key-value pair is only expected in a successful patch twin response message.
                     // Get twin requests will contain the twin version in the payload instead.
-                    version = int.Parse(queryStringKeyValuePairs.Get(VersionTopicKey));
+                    version = int.Parse(queryStringKeyValuePairs.Get(VersionTopicKey), CultureInfo.CurrentCulture);
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs:1242]: InvariantCulture

File: iothub/service/src/DigitalTwin/DigitalTwinsClient.cs [hunk 3]
```diff
@@ -275,7 +275,7 @@ await _internalRetryHandler
                 // No need to deserialize here since the user will deserialize this into their expected type
                 // after this function returns.
                 string responsePayload = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-                int responseStatusCode = int.Parse(response.Headers.GetValues(StatusCodeHeaderKey).FirstOrDefault());
+                int responseStatusCode = int.Parse(response.Headers.GetValues(StatusCodeHeaderKey).FirstOrDefault(), CultureInfo.CurrentCulture);
```
Comment (abhipsaMisra) [inline iothub/service/src/DigitalTwin/DigitalTwinsClient.cs:278]: InvariantCulture

File: iothub/service/src/DigitalTwin/DigitalTwinsClient.cs [hunk 4]
```diff
@@ -364,7 +364,7 @@ await _internalRetryHandler
...
```
Comment (abhipsaMisra) [inline iothub/service/src/DigitalTwin/DigitalTwinsClient.cs:367]: InvariantCulture

File: provisioning/device/src/Transports/Mqtt/MqttLogger.cs [hunk 5]
```diff
@@ -21,7 +22,7 @@ public void Publish(MqttNetLogLevel logLevel, string source, string message, obj
             string formattedMessage = message;
             if (parameters != null && parameters.Length > 0)
             {
-                formattedMessage = string.Format(message, parameters);
+                formattedMessage = string.Format(CultureInfo.CurrentCulture, message, parameters);
```
Comment (abhipsaMisra) [inline provisioning/device/src/Transports/Mqtt/MqttLogger.cs:25]: InvariantCulture

File: iothub/device/src/Transport/Mqtt/MqttLogger.cs (hunk 1 above)
Comment (ngastelum-ms) [reply iothub/device/src/Transport/Mqtt/MqttLogger.cs]: Makes sense, updated

**Pull request merged**